allocator.go

```diff
@@ -78,6 +78,80 @@ type Allocator struct {
 	opts options
 }
 
+type NodeInterface struct{
+	n Node
+	NodeMap map[NodeID]Node
+}
+
+type RangeInterface struct {
+	r        Range
+	RangeMap map[RangeID]Range
+}
```

Comment:
```suggestion
type Nodes map[NodeID]Node

type Ranges map[RangeID]Range
```

Reply:
We can remove the `r` and `n` variables here. Feel free to rename the types as you see fit, I've put in `Nodes/Ranges` as placeholders.